components/BaseMultiselect.vue

```diff
@@ -0,0 +1,284 @@
+  updatedValue = filteredValue;
+  emit("update:modelValue", updatedValue);
+};
+
+const checkByKeyboard = (index: number) => {
+  let updatedValue = [...props.modelValue];
+
+  if (index !== -1 && !updatedValue.includes(props.fileTypes[index])) {
+    updatedValue.push(props.fileTypes[index]);
+  } else {
+    const filteredValue = updatedValue.filter(
+      (value) => value !== props.fileTypes[index]
+    );
+    updatedValue = filteredValue;
+  }
+  emit("update:modelValue", updatedValue);
+};
+</script>
+
+<template>
+  <div class="select">
+    <p class="select__label">{{ label }}</p>
+    <div ref="dropdown" class="select__dropdown">
+      <div class="select__container">
+        <div class="select__selected">
+          <template v-if="modelValue.length > 0">
+            <button
+              v-for="type in modelValue"
+              :key="type"
+              class="select__selected--type"
+              @click="uncheckType(type)"
+            >
+              {{ type }}
+              <UncheckIcon class="select__selected--icon" />
+            </button>
+          </template>
+        </div>
+        <button
+          class="select__button"
+          aria-labelledby="toggle dropdown"
+          aria-haspopup="listbox"
+          :aria-expanded="isDropdownOpen"
+          @click="toggleDropdown"
+          @keyup.esc="isDropdownOpen = false"
+          @keydown.down.prevent="highlightNext()"
+        >
+          <component :is="isDropdownOpen ? SortUp : SortDown" />
+        </button>
+      </div>
+      <ul v-if="isDropdownOpen" class="select__list" role="listbox">
+        <li
+          v-for="(type, index) in fileTypes"
+          :key="type"
+          class="select__listitem"
+          :class="{
+            'select__listitem--highlighted': highlightedIndex === index,
+          }"
+          role="option"
+          @mouseover="highlightedIndex = index"
+        >
+          <input
+            :id="type"
+            class="select__listitem--checkbox"
+            type="checkbox"
+            name="sortType"
+            :value="type"
+            :checked="modelValue.includes(type)"
+            @input="check(type, $event)"
+          />
+          <label
+            :ref="
+              (el) => {
+                if (el !== null) labelsRefs[index] = el;
+              }
+            "
+            :for="type"
+            class="select__listitem--label"
+            :tabindex="highlightedIndex === index ? '0' : '-1'"
+            @keyup.left.prevent="highlightPrev()"
+            @keyup.right.prevent="highlightNext()"
+            @keyup.up.prevent="highlightPrev()"
+            @keyup.down.prevent="highlightNext()"
+            @keyup.enter.prevent="checkByKeyboard(index)"
+            @keyup.space.prevent="checkByKeyboard(index)"
+            @keyup.esc="isDropdownOpen = false"
+          >
+            {{ type }}
+          </label>
+        </li>
+      </ul>
+    </div>
+  </div>
+</template>
```

Comment:
Generally the component is accessible by keyboard, the navigation is not perfect but works, it was quite tricky in this component

Reply:
works really well 👏